lib/core/components/MantraForm.js

```diff
@@ -0,0 +1,63 @@
+import get from 'lodash/get';
+import cloneDeep from 'lodash/cloneDeep';
+import MantraComponent from './MantraComponent';
+import MantraData from './../MantraData';
+import { error } from '../../utils';
+
+/**
+ * Class representing a form component configuration.
+ * @extends MantraComponent
+ */
+export default class MantraForm extends MantraComponent {
+    /**
+     * Creates a MantraForm configuration.
+     * @param {object} payload - Contains the value of the MantraForm properties.
+     */
+    constructor({ name, fields }) {
+        super({ name, mixin: 'MantraForm' });
+        this.fields = fields;
+    };
+
+    /**
+     * Get the config of a MantraForm component.
+     * @param {object} payload - Contains the path and config for MantraData creation.
+     * @returns {MantraData} - The MantraForm component config.
+     */
+    static _getConfig({ path }) {
+        const config = new MantraData({ path, baseClass: MantraForm });
+        
+        try {
+            config.setBase();
+            config.setRecord();
+            config.setAction();
+            config.setComponent();
+            config.setEndpoint();
+        } catch (err) {
+            // Gracefully failing
+            error(err.message);
+        }
+        return config;
+    };
+
+    /**
+     * Gets a MantraForm component object based on a MantraData schema object.
+     * @param {MantraData} data - A MantraData object.
+     * @returns {MantraForm} - A cloned MantraForm component object.
+     */
+    static getComponent(data) {
+        const path = MantraForm._getPath(data);
+        const component = get(data.schema, path);
+        return cloneDeep(component);
+    };
+
+    /**
+     * Gets the path to find the component in the schema.
+     * @param {MantraData} data - A MantraData object.
+     * @returns {string} - Path concatenated by dots. (Empty strings are not added to the path)
+     */
+    static _getPath(data) {
+        const pathInfo = ['actions', data.action.name, data.context];
+        const filteredInfo = pathInfo.filter((str) => str);
```

Comment:
What is the purpose of this filter()? Since the const pathInfo is being set manually with the properties to handle and join(). I see no point in a filter method, could you elaborate pls.

Reply:
This is a temporary filter and it's due to the fact that the "data.context" is not yet implemented. The "data.context" property is set on the constructor of the mantraData to be an empty string. If you join strings and among them, there's an empty string the resultant path will look like: `actions.update.`

That path will be invalid since the "lodash/get" method will not be able to find any information within the "schema" property. So, to avoid it from happening I filtered the array because I wanted to do the least modification possible to the "_getPath" method. My plan was that once the "context" property is fully implemented then that filter will be removed.